xm-commons-function/src/main/java/com/icthh/xm/commons/domain/spec/FunctionSpec.java

```diff
@@ -0,0 +1,112 @@
+package com.icthh.xm.commons.domain.spec;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.icthh.xm.commons.domain.SpecificationItem;
+import com.icthh.xm.commons.domain.enums.FunctionTxTypes;
+import lombok.Data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The {@link FunctionSpec} class.
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"key", "path", "inputSpec", "inputForm", "wrapResult", "anonymous", "txType",
+    "tags", "httpMethods", "validateFunctionInput"})
+@Data
+public class FunctionSpec implements IFunctionSpec, SpecificationItem {
+
+    /**
+     * Unique in tenant function key.
+     */
+    @JsonProperty("key")
+    private String key;
+
+    /**
+     * Unique in tenant HTTP path template.
+     * May contain placeholders for path parameters.
+     * Example: my/rest/service/{id}
+     */
+    @JsonProperty("path")
+    private String path;
+
+    /**
+     * Function input context specification (json-schema for {@code functionInput} arg, see FunctionExecutorService).
+     */
+    @JsonProperty("inputSpec")
+    private String inputSpec;
```

Comment:
pls add outputSpec outputFrom that in entity represents as contextDataSpec contextDataForm

Reply:
updated

Reply:
done